File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.cpp
#include<bits/stdc++.h>
using namespace std;
vector<int> global_numbers;
int global_target,global_answer;
void dfs(int idx,int sum)
{
    if(idx==global_numbers.size()) // 마지막 숫자까지 탐색
    {
        if(sum==global_target)
        {
            global_answer++;
        }
        return;
    }
    dfs(idx+1,sum+global_numbers[idx]);
    dfs(idx+1,sum-global_numbers[idx]);
}
int solution(vector<int> numbers, int target) {
    global_numbers=numbers;
    global_target=target;
    global_answer=0;
    dfs(0,0);
    return global_answer;
}

File: 프로그래머스/2/43165. 타겟 넘버/타겟 넘버.py
global_answer=0
global_numbers=[]
global_target=0

def dfs(idx,total):
    global global_answer,global_numbers,global_target
    
    if idx==len(global_numbers):
        if total==global_target:
            global_answer+=1
        return
    
    dfs(idx+1,total+global_numbers[idx])
    dfs(idx+1,total-global_numbers[idx])

def solution(numbers, target):
    global global_answer,global_numbers,global_target
    
    global_answer=0
    global_numbers=numbers
    global_target=target
    
    dfs(0,0)
    
    return global_answer